### PR TITLE
feat: accessor method for import and derivation method of wallet

### PR DESCRIPTION
**Pull request by marianogoldman**

### CODE REVIEW

File: source/wallet.ts [hunk 1]
```diff
@@ -132,6 +132,16 @@ export class Wallet implements IReadWriteWallet {
 		return this.#attributes.get<Contracts.WalletData>("wallet").primaryKey();
 	}
 
+	/** {@inheritDoc IReadWriteWallet.importMethod} */
+	importMethod(): string {
```
Comment (faustbrian) [inline source/wallet.ts:136]:
```suggestion
	public importMethod(): string {
```

File: source/wallet.ts [hunk 2]
```diff
@@ -132,6 +132,16 @@ export class Wallet implements IReadWriteWallet {
...
+		return this.data().get(WalletData.ImportMethod)!;
+	}
+
+	/** {@inheritDoc IReadWriteWallet.derivationMethod} */
+	derivationMethod(): "bip39" | "bip44" | "bip49" | "bip84" {
```
Comment (faustbrian) [inline source/wallet.ts:141]:
```suggestion
	public derivationMethod(): "bip39" | "bip44" | "bip49" | "bip84" {
```

Comment (faustbrian) [reply source/wallet.ts:141]: Make `"bip39" | "bip44" | "bip49" | "bip84"` a type so it can be properly hinted.

File: source/wallet.mutator.ts [hunk 3]
```diff
@@ -57,6 +57,8 @@ export class WalletMutator implements IWalletMutator {
 			this.#wallet.data().set(WalletData.DerivationPath, path);
 		}
 
+		this.#wallet.data().set(WalletData.ImportMethod, WalletImportMethod.BIP39.MNEMONIC);
```
Comment (marianogoldman) [inline source/wallet.mutator.ts:60]: @faustbrian This is the line we were not sure about earlier. Perhaps it doesn't really belong in here, but as soon as I added it all the wallets used in tests started to have a valid import method set. Perhaps the issue is with how the wallets are built (for tests, at least). Most of the examples are built by just doing:
```
	subject = new Wallet(uuidv4(), {}, profile);
```
And then updated using:
```
	await subject.mutator().coin("ARK", "ark.devnet");
	await subject.mutator().identity(identity.mnemonic);
```
or similar.
I think setting the identity (or address or ext public key, and even coin) are things that should be immutable for a wallet. So I don't understand the purpose of the mutator (other than setting alias or avatar). I'm not sure what was the idea of introducing the mutator (perhaps just for DRY).
Having said all that, for tests perhaps we could change all creation of wallets to use the wallet factory (which already properly sets the import and derivation methods.

Comment (faustbrian) [reply source/wallet.mutator.ts]: The mutator just exists to keep things DRY. Haven't had the time yet to rework more parts.

**Pull request merged**